journal.py

```diff
@@ -16,137 +13,99 @@
-            db.commit()
-        request.db.close()
-
-
-def write_entry(request):
-    """write a single entry to the database"""
-    title = request.params['title']
-    text = request.params['text']
-    created = datetime.today()
-    request.db.cursor().execute(INSERT_ENTRY, (title, text, created))
-    # write to a separate file
-    return {}
-
-
-def update(request, identification):
-    title = request.params['title']
-    text = request.params['text']
-    created = datetime.today()
-    request.db.cursor().execute(UPDATE_ENTRY, (title, text, created, identification))
-    return {}
 
+    def __repr__(self):
+        return u"{}: {}".format(self.__class__.__name__, self.title)
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def by_id(cls, id, session=None):
+        if session is None:
+            session = DBSession
+        return DBSession.query(cls).filter(cls.id == id).one()
+
+    @classmethod
+    def delete_by_id(cls, id, session=None):
+        if session is None:
+            session = DBSession
+        entry = DBSession.query(cls).filter(cls.id == id).one()
+        DBSession.delete(entry)
+
+    @classmethod
+    def from_request(cls, request, session=None):
+        if session is None:
+            session = DBSession
+        title = request.params.get('title', None)
+        text = request.params.get('text', None)
+        created = datetime.today()
+        new_entry = cls(title=title, text=text, created=created)
+        DBSession.add(new_entry)
+        return new_entry
+
+    @classmethod
+    def delete_all(cls, session=None):
+        if session is None:
+            session = DBSession
 
 @view_config(route_name='add', request_method='POST', renderer="json")
 def add_entry(request):
     if request.authenticated_userid:
         if request.method == 'POST':
+            new = None
             try:
-                write_entry(request)
+                Entry.from_request(request)
+                new = DBSession.query(Entry).order_by(Entry.id.desc()).first()
             except psycopg2.Error:
-                # this will catch any errors generated by the database
                 return HTTPInternalServerError
-            cursor = request.db.cursor()
-            cursor.execute(DB_MOST_RECENT, [])
-            keys = ('id', 'title', 'text', 'created')
-            entry = dict(zip(keys, cursor.fetchone()))
-            entry['text'] = markdown.markdown(entry['text'], extensions=('codehilite', 'fenced_code'))
-            entry['created'] = entry['created'].strftime('%b. %d, %Y')
+            entry = {
+                'id': new.id,
+                'title': new.title,
+                'text': markdown.markdown(new.text, extensions=('codehilite', 'fenced_code')),
+                'created': new.created.strftime('%b. %d, %Y')
+                }
```

Comment:
Why not simply return the entry object you get back from the database?  Alternatively, if you're trying to convert your entry to a JSON-compatible dict, you could implement an instance method on the Entry class to handle that.  It'd be cleaner than this.